webextension/background.js

```diff
@@ -573,14 +615,76 @@ async function onNavigationCompleted(navDetails) {
     return;
   }
 
-  // When the page has loaded, maybe prompt the user.
-  if (await shouldPromptUser(navDetails)) {
-    gCurrentlyPromptingTab = {id: tabId, url};
-    await updatePageActionIcon(tabId);
-    await browser.pageAction.show(tabId);
-    popupPageAction(tabId);
-    Config.onUserPrompted(new URL(url).host);
+  // If the user was already in mid-prompt on this tab,
+  // reopen the prompt.
```

Comment:
The idea there was that if they navigate back to the same URL (refreshing the page etc), then don't wait the extra 5-10 seconds to prompt them again, just pop up as soon as the page has loaded.

But maybe that's a flawed assumption? Maybe if they refresh the page or navigate to another URL on the same domain instead of completing the prompt, we should just treat it as them dismissing the prompt?

Reply:
Let's add back the random delay before prompting in this case.